internal/bminventory/inventory.go

```diff
@@ -1660,6 +1667,28 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
 	return b.v2UpdateClusterInternal(ctx, params, NonInteractive)
 }
 
+func alignClusterPlatformParams(params installer.V2UpdateClusterParams, cluster *common.Cluster) error {
+	paramsUserManagedNetworking := params.ClusterUpdateParams.UserManagedNetworking
+
+	// Attempts to set UserManagedNetworking=false and Platform=none or UserManagedNetworking=true and Platform=baremetal
+	// will result StatusBadRequest error
+	if paramsUserManagedNetworking != nil && params.ClusterUpdateParams.Platform != nil {
+		platformType := *params.ClusterUpdateParams.Platform.Type
+		if !*paramsUserManagedNetworking && platformType == models.PlatformTypeNone || *paramsUserManagedNetworking && models.PlatformTypeBaremetal == platformType {
+			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform with user-managed-networking %v", platformType, *paramsUserManagedNetworking))
+		}
+	}
+
+	if swag.BoolValue(paramsUserManagedNetworking) && *cluster.Platform.Type == models.PlatformTypeBaremetal {
+		params.ClusterUpdateParams.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}
+	} else if params.ClusterUpdateParams.Platform != nil && *params.ClusterUpdateParams.Platform.Type == models.PlatformTypeNone {
+		params.ClusterUpdateParams.UserManagedNetworking = swag.Bool(true)
+	} else if paramsUserManagedNetworking != nil && !*paramsUserManagedNetworking && *cluster.Platform.Type == models.PlatformTypeNone {
```

Comment:
This case (UMN=false, platform=None) should return bad request

Reply:
Didn't we said we want to keep the current behavior?
If the user set the UMN=true that resulting platform=none, now updating UMN=false will return bad request.
In this case we will need to also change the UI behavior?